### PR TITLE
Show all file content in iframes

### LINKED ISSUE #142: Lazily load csv and text file content from direct url.
Currently, for rendering file contents, we have two different techniques.

For html and images, we use `<iframe>` or `` respectively with `src=url...` tag. This means the content is loaded separately by the browser 

This is useful because the url responses with appropriate cache headers, so browser can cache and do conditional GETs and avoid reloading as a user switches between files.

For csv and text, we render the content directly inline, as part of the entire page html.  This makes caching the file contents not really possible, as they are mixed in with all the rest. It also makes the template rendering slower and more complex, as its embedding a lot of html in the rendering.

Instead, we could split out the generation of the html representation of csv and plain text into a separate urls and link to that, just like we do for html iframes and image tags.  This would give us the benefits of cacheability for this content too, and make the main view html simpler and faster.

Given the renderable content of csv's and plain text is actually html, there are two ways to do this:

1. Use iframe. This is simple, but iframes are weird beasts to control. We use them for html outputs mainly because we want to isolate from them from the css/js in the container page.  Doing this would mean we would need to bundle the css for tables and plain text (e.g. syntax highlighting) in the embedable html content too.

2. Use HTMX's `hx-get` and `hx-trigger=load` to lazily load them, and just inject the html into the page at the right place.  We control the html for these, so we don't need to worry about isolation as much.

### PR DESCRIPTION
**Pull request by bloodearnest**
Fixes #142

### CODE REVIEW

File: tests/integration/views/test_helpers.py [hunk 1]
```diff
@@ -1,16 +1,70 @@
...
+    "suffix,mimetype",
+    [
+        (".html", "text/html"),
+        (".csv", "text/html"),
+        (".png", "image/png"),
+        (".txt", "text/html"),
+    ],
+)
+def test_serve_file_rendered(tmp_path, suffix, mimetype):
+    rf = RequestFactory()
```
Comment (rebkwok) [inline tests/integration/views/test_helpers.py:19]: pytest-django has a `rf` fixture that could be used here instead (and in the other tests in this file)

File: tests/integration/views/test_helpers.py [hunk 2]
```diff
@@ -1,16 +1,70 @@
...
+@pytest.mark.parametrize(
+    "filename,mimetype",
+    [
+        ("foo.html", "text/html"),
+        ("foo.csv", "text/html"),
+        ("foo.png", "image/png"),
+        ("foo.txt", "text/html"),
+    ],
+)
+def test_serve_file_filename(tmp_path, filename, mimetype):
```
Comment (rebkwok) [inline tests/integration/views/test_helpers.py]: very minor nit; the test is testing the headers, not the filename

File: airlock/views/helpers.py [hunk 3]
```diff
@@ -37,40 +44,69 @@ def get_release_request_or_raise(user, request_id):
...
+        @functools.wraps(func)
+        def wrapper(abspath, suffix):
+            context = func(abspath, suffix)
+            return SimpleTemplateResponse(template, context)
+
+        wrapper.template_path = template_path
+
+        return wrapper
+
+    return decorator
```
Comment (rebkwok) [inline airlock/views/helpers.py:74]: Neat

**Review by rebkwok — APPROVED** [review]

File: tests/integration/views/test_helpers.py (hunk 2 above)
Comment (bloodearnest) [reply tests/integration/views/test_helpers.py]: Ah, right, yes. It's testing it with the filename parameter though.

File: tests/integration/views/test_helpers.py (hunk 1 above)
Comment (bloodearnest) [reply tests/integration/views/test_helpers.py:19]: Ah, yes, I'd forgotten about that, will add.

File: tests/integration/views/test_helpers.py (hunk 2 above)
Comment (bloodearnest) [reply tests/integration/views/test_helpers.py]: Also, this didn't make it into the final commit, was intermediate, so already gone!

**Pull request merged**